modules/core/src/core/animation-loop.js

```diff
@@ -234,6 +238,63 @@ export default class AnimationLoop {
     return this.props.onInitialize(...args);
   }
 
+  onRenderFrame(...args) {
```

Comment:
I don't like having all this custom code in the main render loop for an experimental feature. What could go in here is an option to call onRender more than once and pass the index of the render to the callback. The VR setup could be a utility `onRender` function that's defined elsewhere.